svm/src/transaction_processor.rs

```diff
@@ -2178,10 +2179,14 @@ mod tests {
 
             let mut error_counters = TransactionErrorMetrics::default();
             let batch_processor = TransactionBatchProcessor::<TestForkGraph>::default();
-            let nonce = Some(NonceInfo::new(
-                *fee_payer_address,
-                fee_payer_account.clone(),
-            ));
+
+            let durable_nonce = DurableNonce::from_blockhash(&last_blockhash);
+            let mut nonce_info = NonceInfo::new(*fee_payer_address, fee_payer_account.clone());
+            nonce_info
+                .try_advance_nonce(durable_nonce, lamports_per_signature)
+                .unwrap();
```

Comment:
I imagine you're advancing here so that we make sure rollback accounts is picking up the new nonce data? Do you mind adding a comment explaining that's why we are advancing? Because without this advancement, the test still passes so it's not immediately clear why it's important.

Reply:
this was just a mistake of imagination editing the edited version of this test, since the previous version of the code advanced the nonce in `validate_transaction_fee_payer()` but the current doesnt